key_test.go

```diff
@@ -0,0 +1,68 @@
+package shell
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cheekybits/is"
+)
+
+func TestKeyGen(t *testing.T) {
+	is := is.New(t)
+	s := NewShell(shellUrl)
+
+	key, err := s.KeyGen(context.Background(), "testKey", KeyGen.Type("ed25519"), KeyGen.Size(2048))
+	is.Nil(err)
+
+	is.Equal(key.Name, "testKey")
+	is.NotNil(key.Id)
+
+	_, err = s.KeyRm(context.Background(), "testKey")
+	is.Nil(err)
+}
+
+func TestKeyList(t *testing.T) {
+	is := is.New(t)
+	s := NewShell(shellUrl)
+
+	keys, err := s.KeyList(context.Background())
+	is.Nil(err)
+
+	is.Equal(len(keys), 1)
+	is.Equal(keys[0].Name, "self")
+	is.NotNil(keys[0].Id)
```

Comment:
We should verify the KeyList works with multiple return values. Try inserting a few keys and then doing a KeyList (then remove them so we leave the state as we found it).